modules/gateways/veritrans.php

```diff
@@ -360,10 +360,24 @@ function veritrans_link($params)
         $snapToken = $snap_transaction->token;
         $redirect_url = $snap_transaction->redirect_url;
         // error_log(" ############# TOKEN ::: ".$snapToken);
+
+        // save snap redirect url to invoice notes
+        $command = 'UpdateInvoice';
+        $postData = array(
+            'invoiceid' => $invoiceId,
+            'notes' => $redirect_url
+        );
+        $results = localAPI($command, $postData);
+
     } catch (Exception $e) {
         // error_log('Caught exception: ',  $e->getMessage(), "\n");
-        if (preg_match('/utilized/',$e->getMessage()) ){
-            return "<h4> Awaiting Your Payment </h4>";
+        if (preg_match('/utilized|digunakan/',$e->getMessage()) ){
+            $command = 'GetInvoice';
+            $postData = array(
+                'invoiceid' => $invoiceId
+            );
+            // Display payment instruction, prevent payment button from being shown
+            return "Invoice ID has been created on Midtrans previously. Please try to check your email for the previous payment instruction details or open link on Invoice Notes.";
```

Comment:
Why does the message tell customer to manually check the URL on invoice notes? That's not an optimal UX. can't the message just show it as link directly and asking customer to visit the link?